app/assets/javascripts/discourse/app/components/modal/login.js

```diff
@@ -0,0 +1,301 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { SECOND_FACTOR_METHODS } from "discourse/models/user";
+import { ajax } from "discourse/lib/ajax";
+import { findAll } from "discourse/models/login-method";
+import { areCookiesEnabled } from "discourse/lib/utilities";
+import { wavingHandURL } from "discourse/lib/waving-hand-url";
+import { schedule } from "@ember/runloop";
+import cookie, { removeCookie } from "discourse/lib/cookie";
+import { isEmpty } from "@ember/utils";
+import I18n from "I18n";
+import { escape } from "pretty-text/sanitizer";
+
+export default class Login extends Component {
+  @service dialog;
+  @service siteSettings;
+
+  @tracked loggingIn = false;
+  @tracked loggedIn = false;
+  @tracked showLoginButtons = true;
+  @tracked showSecondFactor = false;
+  @tracked loginPassword = "";
+  @tracked loginName = "";
+  @tracked flash = this.args.model?.flash;
+  @tracked flashType = this.args.model?.flashType;
+  @tracked canLoginLocal = this.siteSettings.enable_local_logins;
+  @tracked
+  canLoginLocalWithEmail = this.siteSettings.enable_local_logins_via_email;
+  @tracked secondFactorMethod = SECOND_FACTOR_METHODS.TOTP;
+  @tracked securityKeyCredential;
+  @tracked otherMethodAllowed;
+  @tracked secondFactorRequired;
+  @tracked backupEnabled;
+  @tracked totpEnabled;
+  @tracked showSecurityKey;
+  @tracked securityKeyChallenge;
+  @tracked securityKeyAllowedCredentialIds;
+  @tracked secondFactorToken;
+
+  constructor() {
+    super(...arguments);
+    if (this.args.model?.isExternalLogin) {
+      this.externalLogin(this.args.model.externalLoginMethod, {
+        signup: this.args.model.signup,
+      });
+    }
+  }
```

Comment:
Once this PR lands I would love to come back in and refactor this block. We render the `Login` modal in multiple places to then immediately redirect to an external auth service. I would like to skip the intermediate `LoginModal` step.